.circleci/config.yml

```diff
@@ -0,0 +1,35 @@
+version: 2.1
+
+jobs:
+  build:
+    docker:
+      - image: mcr.microsoft.com/dotnet/sdk:7.0
+    steps:
+      - run:
+          name: Install ssh client
+          command: apt-get update && apt-get install -y openssh-client
+      - checkout
+      - run:
+          name: Restore packages
+          command: dotnet restore
+      - run:
+          name: Build
+          command: dotnet build -c Release
+      - run:
+          name: Test
+          command: dotnet test "Kumi.Tests/Kumi.Tests.csproj" -c Release --no-restore --no-build --logger "trx"
+      - run:
+          name: Upload results
+          when: always
+          command: |
+            export PATH="$PATH:/root/.dotnet/tools"
+            dotnet tool install -g trx2junit
+            ls -la Kumi.Tests
+            trx2junit Kumi.Tests/TestResults/*.trx
+      - store_test_results:
+          path: Kumi.Tests/TestResults
```

Comment:
Is there a means we can upload code coverage, or has previous endeavors when I did `this exact same thing` with compose provided that we hold trauma from ever trying.

Reply:
I did include the test coverage within the artifacts in CircleCI, so it's possible to upload the test coverage somewhere else. I haven't bothered though since I thought if we were to go down that path it would be better if we did so in another PR or something along the lines.

The artifacts uploaded are both in the `.xml` and `.trx` format, so anything that supports those formats should work well.

Reply:
That's the same presumption I made when I first tried doing the same thing with CircleCI back when Composer was still alive. `.trx` files are not coverage files, as they purely serve to provide results as to whether or not the tests actually passed altogether. [It is a creation of Microsoft's as part of Visual Studio's own testing system](https://learn.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022#:~:text=Visual%20Studio%20Test%20Results%20File%20(trx)), as implemented into other similar IDEs as well.

What you would need to do is use an entirely seperate tooling to generate such files for test coverage such as [Coverlet](https://github.com/coverlet-coverage/coverlet), but my experiences with coercing them to work with CircleCI was the exact nightmare that made me never want to. 

The reason?

Coverlet in particular only works on `Windows`, and is impossible to effectively use on any other platform, and CircleCI's Windows containers are fucking awful.

Reply:
[This entire so-on-so guide proves why](https://circleci.com/docs/using-windows/), and I'm unsure if Coverlet will even like being automatically used in this regard. It's also the only real option available as `NCover` is commercial.

Reply:
I could attempt to make Coverlet work under CircleCI, but I think it would be better if we did that within another PR since I feel like it'd be quite a significant change (and I'd also need to create another branch to test my changes anyway)